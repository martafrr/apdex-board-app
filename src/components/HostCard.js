import React from "react";

const HostCard = ({name, hostsApp}) => {
    const hostCard = {
        background: '#d6d7dc',
        color: '#2c3437',
        fontSize: '14px',
        // TODO: change to dinamic height
        height: '150px',
        padding: '5px 11px',
        margin: '10px',
    }

    const apps = hostsApp.map(app =>
        <div key={app.name}>
            <p>{app.apdex} - {app.name}</p>
        </div>
        
    );
    return (
        <div style={hostCard}>
            <h3>{name}</h3>
            {apps}
        </div>
    )
}

export default HostCard;
