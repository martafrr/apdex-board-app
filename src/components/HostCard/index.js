import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import AppModal from '../AppModal';

const HostCard = ({ hostsApp, hostName }) => {
    const [modalOpen, setModalOpen] = useState(false)
    const [modalAppInfo, setModalAppInfo] = useState(null)
    const [appsIndex, setAppsIndex] = useState(null)

    const handleClickApp = (e) => {
        const index = e.target.value;
        const clickedApp = hostsApp[index];
        setModalOpen(true);
        setModalAppInfo(clickedApp);
        setAppsIndex(index);
    }
    const handleOnClose = () => {
        setModalOpen(false);
    }

    const hostCardStyles = {
        background: '#d6d7dc',
        fontSize: '14px',
        height: '280px',
        padding: '5px 11px',
        margin: '10px',
    }
    const hostNameStyles = {
        color: '#2c3437',
    }
    const appNameStyles = {
        background: 'none',
        border: 'none',
        fontSize: '13px',
        margin: '7px auto',
        textAlign: 'left',
    }

    const top5Apps = hostsApp.map((app, index) => {
        if(index < 5) {
            return (
                <div key={app.name}>
                    <button 
                        className='openModalButton'
                        style={appNameStyles} 
                        value={index} 
                        onClick={handleClickApp}
                        data-test="open-modal-button"
                    >
                        {app.apdex} - {app.name}
                    </button>
                </div>
            )
        }
        return null;
    });

    return (
        <div style={hostCardStyles} data-test="host-card">
            <Link style={hostNameStyles} to={'/' + hostName}>
                <h3>{hostName}</h3>
            </Link>

            {top5Apps}

            {modalOpen && 
                <AppModal 
                    isOpen={modalOpen}
                    hostName={hostName}
                    appsIndex={appsIndex} 
                    appInfo={modalAppInfo} 
                    onClose={handleOnClose}
                />}
        </div>
    );
}

HostCard.propTypes = {
    hostsApp: PropTypes.array,
    hostName: PropTypes.string,
}

export default HostCard;
