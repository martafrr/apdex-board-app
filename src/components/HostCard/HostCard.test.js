import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import HostCard from './index';

describe('HostCard Component', () => {
    let component;
    const mockProps = {
        name: 'name',
        hostsApp: ['host 1', 'host 2', 'host 3']
    }

    beforeEach(() => {
        component = shallow(<HostCard {...mockProps} />);

    });
    it('matches the snapshot', () => {
        component = create(
            <BrowserRouter>
                <HostCard {...mockProps} />
            </BrowserRouter>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
