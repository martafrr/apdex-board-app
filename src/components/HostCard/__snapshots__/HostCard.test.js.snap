// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostCard Component matches the snapshot 1`] = `
<div
  data-test="host-card"
  style={
    Object {
      "background": "#d6d7dc",
      "fontSize": "14px",
      "height": "280px",
      "margin": "10px",
      "padding": "5px 11px",
    }
  }
>
  <a
    href="/undefined"
    onClick={[Function]}
    style={
      Object {
        "color": "#2c3437",
      }
    }
  >
    <h3 />
  </a>
  <div>
    <button
      className="openModalButton"
      data-test="open-modal-button"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "none",
          "fontSize": "13px",
          "margin": "7px auto",
          "textAlign": "left",
        }
      }
      value={0}
    >
       - 
    </button>
  </div>
  <div>
    <button
      className="openModalButton"
      data-test="open-modal-button"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "none",
          "fontSize": "13px",
          "margin": "7px auto",
          "textAlign": "left",
        }
      }
      value={1}
    >
       - 
    </button>
  </div>
  <div>
    <button
      className="openModalButton"
      data-test="open-modal-button"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "none",
          "fontSize": "13px",
          "margin": "7px auto",
          "textAlign": "left",
        }
      }
      value={2}
    >
       - 
    </button>
  </div>
</div>
`;
